src/details/ArborX_DetailsBoundingVolumeHierarchyImpl.hpp

```diff
@@ -381,6 +381,9 @@ BoundingVolumeHierarchyImpl<DeviceType>::queryDispatch(
   using Access = Traits::Access<Predicates, Traits::PredicatesTag>;
   auto const n_queries = Access::size(predicates);
 
+  bool const throw_if_buffer_optimization_fails = (buffer_size < 0);
```

Comment:
It should be documented not just in an internal file what a negative value should imply.

Reply:
As we expect API to change, we don't document this or expose it to users yet.

Reply:
Hyrum's law still applies. :wink:

Reply:
OK, I invoke "out of scope for this PR".